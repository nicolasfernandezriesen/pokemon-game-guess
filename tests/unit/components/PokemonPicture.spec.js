import {shallowMount} from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemonId: 1,
                showPokemon: false
            }
        })     
    })
    test('it should match with the snapshot', () => {
        //Arrange
        //Act
        //Assert
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('it should show the pokemon shadow img', () => {
        //Arrange
        //Act
        let [img1, img2] = wrapper.findAll('img');

        //Assert
        expect(img1.exists()).toBeTruthy();
        expect(img2).toBe(undefined);
        expect(img1.classes('hidden-pokemon')).toBe(true);
    })

    test('it should show the pokemon img', () => {
        //Arrange
        let wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemonId: 1,
                showPokemon: true
            }
            
        });
        
        //Act
        let img = wrapper.find('img');

        //Assert
        expect(img.exists()).toBeTruthy();
        expect(img.classes('fade-in')).toBe(true);
    })

    test('it should match de src', () => {
        //Arrange
        let wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemonId: 1,
                showPokemon: true
            }
            
        });

        //Act
        let img = wrapper.find('img');

        //Assert
        expect(img.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg');
    })
})