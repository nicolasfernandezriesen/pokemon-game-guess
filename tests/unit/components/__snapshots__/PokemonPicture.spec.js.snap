// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture it should match with the snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon"></div>`;
